scripts/globals/spells/bluemagic/pinecone_bomb.lua

```diff
@@ -0,0 +1,48 @@
+---------------------------------------------
+--  Pinecone Bomb
+--  Description: Single target damage with sleep.
+-- Todo add damage to log, atm if it shows damage it wont sleep the mob
+---------------------------------------------
+require("scripts/globals/status")
+require("/scripts/globals/bluemagic")
+require("scripts/globals/msg");
+---------------------------------------------
+
+function onMagicCastingCheck(caster,target,spell)
+    return 0
+end
+
+function onSpellCast(caster,target,spell)
+    local params = {}
+    local multi = 1.25
+    if(caster:hasStatusEffect(dsp.effect.AZURE_LORE)) then
+        multi = multi + 0.50
+    end
+        params.tpmod = TPMOD_CRITICAL
+        params.dmgtype = DMGTYPE_PIERCE
+        params.scattr = SC_DARK
+        params.numhits = 1
+        params.multiplier = multi
+        params.tp150 = 1.25
+        params.tp300 = 1.25
+        params.azuretp = 1.25
+        params.duppercap = 8
+        params.str_wsc = 0.0
+        params.dex_wsc = 0.0
+        params.vit_wsc = 0.0
+        params.agi_wsc = 0.0
+        params.int_wsc = 0.2
+        params.mnd_wsc = 0.0
+        params.chr_wsc = 0.0
+    damage = BluePhysicalSpell(caster, target, spell, params)
+    damage = BlueFinalAdjustments(caster, target, spell, damage, params)
+
+    if (damage > 0) then
+        local typeEffect = dsp.effect.SLEEP_I
+        target:delStatusEffect(typeEffect)
```

Comment:
does it always overwrite even if the effect its replacing is stronger? because thats what you have happening here,

Reply:
Well it would overwrite becuase it does damage 1st then a sleep effect. Its the only physical sleep spell in the game

Reply:
that just makes me realize the delstatus was even more redundant than I thought it was. since damage wakes you and this only applies sleep if you were hit/damaged by it that means there will never be a status to delete when delStatusEffect gets called.

if you do zero damage but landed it, will retail sleep? this script won't because checking for dmg > 0

probably want to check for hit instead of dmg value but if it must be > 0 dmg then we do what shockwave does. I recall there being a core issue with the order of operation, if thats not sorted out already then no script will work for this yet anyway. I *think* your other blue spell is ok but ima wait and let someone else look at that